packages/attribute-registry-process/src/services/readModelService.ts

```diff
@@ -190,4 +191,49 @@ export const readModelService = {
       "data.code": code,
     });
   },
+  async getAttributeByCodeAndName(
+    code: string,
+    name: string
+  ): Promise<WithMetadata<Attribute> | undefined> {
+    return getAttribute({
+      "data.code": code,
+      "data.name": {
+        $regex: `^${name}$$`,
+        $options: "i",
+      },
+    });
+  },
+  async getTenantById(
+    tenantId: string
+  ): Promise<WithMetadata<Tenant> | undefined> {
+    const data = await tenants.findOne(
+      { "data.id": tenantId },
+      { projection: { data: true, metadata: true } }
+    );
+
+    if (data) {
+      const result = z
+        .object({
+          metadata: z.object({ version: z.number() }),
+          data: Tenant,
+        })
+        .safeParse(data);
+
+      if (!result.success) {
+        logger.error(
+          `Unable to parse tenant item: result ${JSON.stringify(
+            result
+          )} - data ${JSON.stringify(data)} `
+        );
+
+        throw ErrorTypes.GenericError;
+      }
+
+      return {
+        data: result.data.data,
+        metadata: { version: result.data.metadata.version },
+      };
+    }
+    return undefined;
+  },
```

Comment:
I generally try to avoid too much nesting, if not at all, it tends to make the code less readable.
For example, in this case: in order to understand that the function return `undefined` if there is no data, we need to make a "jump" from line 214 to line 237.

I would suggest a [guard clause](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/).

Let me know what you think!

```suggestion
 if (!data) {
      return undefined;
    }

    const result = z
      .object({
        metadata: z.object({ version: z.number() }),
        data: Tenant,
      })
      .safeParse(data);

    if (!result.success) {
      logger.error(
        `Unable to parse tenant item: result ${JSON.stringify(
          result
        )} - data ${JSON.stringify(data)} `
      );

      throw ErrorTypes.GenericError;
    }

    return {
      data: result.data.data,
      metadata: { version: result.data.metadata.version },
    };
```